### PR TITLE
ENG-18825: StreamBlock changes:

### PR DESCRIPTION
**Pull request by manjujames**
- Accept sequence number for releaseTo > lastSequenceNo
- New method to get a read only buffer with refcount incremented

### CODE REVIEW

File: src/frontend/org/voltdb/export/StreamBlock.java [hunk 1]
```diff
@@ -171,6 +170,11 @@ public BBContainer unreleasedContainer() {
         return getRefCountingContainer(m_entry.getData().slice().asReadOnlyBuffer());
     }
 
+    public ByteBuffer getReadOnlyBufferWithRefCount() {
```
Comment (rdykiel) [inline src/frontend/org/voltdb/export/StreamBlock.java]: Maintenance: we need comments explaining the various flavors for these methods and what are their intended use

File: src/frontend/org/voltdb/export/StreamBlock.java [hunk 2]
```diff
@@ -142,9 +142,8 @@ long unreleasedRowCount()
     public void releaseTo(long releaseSequenceNumber)
     {
         assert(releaseSequenceNumber >= m_startSequenceNumber);
-        m_releaseOffset = (int)(releaseSequenceNumber - m_startSequenceNumber);
-        // if it is fully released, we will discard the block
-        assert(m_releaseOffset < (m_rowCount - 1));
+        m_releaseOffset = releaseSequenceNumber >= lastSequenceNumber() ?
+                (m_rowCount - 1) : (int)(releaseSequenceNumber - m_startSequenceNumber);
```
Comment (rdykiel) [inline src/frontend/org/voltdb/export/StreamBlock.java]: Comment why we accept a released seqno > last seqno

Comment (manjujames) [reply src/frontend/org/voltdb/export/StreamBlock.java]: Now that we push acked buffers as well, if the buffer is fully acked, we need to set the releasedSeqNo accordingly.

File: src/frontend/org/voltdb/export/StreamBlock.java (hunk 1 above)
Comment (n3world) [reply src/frontend/org/voltdb/export/StreamBlock.java]: I don't think you need this method it just makes the api a bit more complicated. You could instead have PollBlock extend DBBDelegateContainer and remove the discard implementation

**Pull request merged**